package/Shaders/ISVolumetricLightingGenerateCS.hlsl

```diff
@@ -69,84 +68,63 @@ cbuffer PerTechnique : register(b0)
 }
 
 [numthreads(32, 32, 1)] void main(uint3 dispatchID : SV_DispatchThreadID) {
-	const float3 StepCoefficients[] = { { 0, 0, 0 },
-		{ 0, 0, 1.000000 },
-		{ 0, 1.000000, 0 },
-		{ 0, 1.000000, 1.000000 },
-		{ 1.000000, 0, 0 },
-		{ 1.000000, 0, 1.000000 },
-		{ 1.000000, 1.000000, 0 },
-		{ 1.000000, 1.000000, 1.000000 } };
-
-	float3 normalizedCoordinates = dispatchID.xyz / TextureDimensions.xyz;
+	const float3 StepCoefficients[] = {
+		{ 0, 0, 0 },
+		{ 0, 0, 0.001 },
+		{ 0, 0.001, 0 },
+		{ 0, 0.001, 0.001 },
+		{ 0.001, 0, 0 },
+		{ 0.001, 0, 0.001 },
+		{ 0.001, 0.001, 0 },
+		{ 0.001, 0.001, 0.001 }
+	};
+
+	float3 normalizedCoordinates = dispatchID.xyz * rcp(TextureDimensions.xyz);
 	float2 uv = normalizedCoordinates.xy;
 	uint eyeIndex = Stereo::GetEyeIndexFromTexCoord(uv);
-	float3 depthUv = Stereo::ConvertFromStereoUV(normalizedCoordinates, eyeIndex) + 0.001 * StepCoefficients[IterationIndex].xyz;
-	float depth = InverseRepartitionTex.SampleLevel(InverseRepartitionSampler, depthUv.z, 0).x;
+	float3 depthUv = Stereo::ConvertFromStereoUV(normalizedCoordinates, eyeIndex) + StepCoefficients[IterationIndex];
+	float depth = InverseRepartitionTex.SampleLevel(InverseRepartitionSampler, depthUv.z, 0);
 	float4 positionCS = float4(2 * depthUv.x - 1, 1 - 2 * depthUv.y, depth, 1);
 
-	float4 positionWS = mul(transpose(CameraViewProjInverse[eyeIndex]), positionCS);
-	positionWS /= positionWS.w;
+	float4 positionWS = mul(CameraViewProjInverse[eyeIndex], positionCS);
+	positionWS *= rcp(positionWS.w);
 
-	float4 positionCSShifted = mul(transpose(CameraViewProj[eyeIndex]), positionWS);
-	positionCSShifted /= positionCSShifted.w;
+	float4 positionCSShifted = mul(CameraViewProj[eyeIndex], positionWS);
+	positionCSShifted *= rcp(positionCSShifted.w);
 
 	float shadowMapDepth = positionCSShifted.z;
 
-	float shadowContribution = 1;
+	bool noShadow = true;
 	if (EndSplitDistances.z >= shadowMapDepth) {
```

Comment:
@coderabbitai The "if (EndSplitDistances.z >= shadowMapDepth) {" and code within this block I believe appear in the utility shader. Please confirm 1) whether the code exists and 2) whether the optimizations to this block might make sense in the other code. If so, please create an unassigned issue.